sqlconnect/internal/snowflake/config.go

```diff
@@ -58,11 +70,17 @@ func (c Config) ConnectionString() (dsn string, err error) {
 			return "", fmt.Errorf("parsing private key: %w", err)
 		}
 		sc.PrivateKey = privateKey
+	} else if c.UseOAuth {
+		sc.Authenticator = gosnowflake.AuthTypeOAuth
+		sc.Token = c.OAuthToken
+		sc.Port = 443
+		sc.Protocol = "https"
 	}
 
 	if c.KeepSessionAlive {
-		valueTrue := "true"
-		sc.Params["client_session_keep_alive"] = &valueTrue
+		// valueTrue := "true"
+		// sc.Params["client_session_keep_alive"] = &valueTrue
+		sc.KeepSessionAlive = true
```

Comment:
Based on [this](https://github.com/snowflakedb/gosnowflake/issues/738#:~:text=So%20if%20i,driver%20connection%20close) I would argue that this change is not desirable. Why do we need `KeepSessionAlive`?
```suggestion
		valueTrue := "true"
		sc.Params["client_session_keep_alive"] = &valueTrue
```